File: diamond.js
function diamondPattern(floor) {
  let space = floor - 1;

  for (let q = 1; q <= floor; q++) {
    let line = "";
    for (let w = 1; w <= space; w++) {
      line += " ";
    }
    space--;

    for (let w = 1; w <= 2 * q - 1; w++) {
      line += "*";
    }

    console.log(line);
  }

  space = 1;
  for (let q = floor - 1; q >= 1; q--) {
    let line = "";
    for (let w = 1; w <= space; w++) {
      line += " ";
    }
    space++;

    for (let w = 1; w <= 2 * q - 1; w++) {
      line += "*";
    }

    console.log(line);
  }
}

const floor = parseInt(prompt("number of floor you wish to enter :"));
diamondPattern(floor);

File: diamond.py
r = int(input(" i want you to please to enter the number of rows :"))
space = r - 1

for w in range(1, r + 1):
    for q in range(1, r - w + 1):
        print(" ", end="")
    for q in range(1, 2 * w):
        print("*", end="")
    print()

for q in range(r, 0, -1):
    for w in range(1, space - 2):
        print(" ", end="")
    space += 1
    for w in range(1, q + 1):
        print("*", end="")
    for w in range(1, q):
        print("*", end="")
    print()
